controller_manager/controller_manager/spawner.py

```diff
@@ -43,21 +43,65 @@ class bcolors:
     UNDERLINE = '\033[4m'
 
 
-def wait_for_controller_manager(node, controller_manager, timeout_duration):
-    def full_name(n):
-        return n[1] + ('' if n[1].endswith('/') else '/') + n[0]
+def first_match(iterable, predicate):
+    return next(n for n in iterable if predicate(n), None)
```

Comment:
The problem is here!
```suggestion
    return next((n for n in iterable if predicate(n)), None)
```